### DISCUSSION TITLE
Is there a way to override remark-cli config based on path?

### DISCUSSION THREAD

**Discussion by silvenon** [unifiedjs/unified#126] [Q&A] [answered]
You can tweak ESLint configuration using `overrides` based on path, for example:

```js
module.exports = {
  // base ESLint rules
  overrides: [
    {
      files: "*.ts",
      rules: {
        // some TypeScript-specific rules
      },
    },
  ],
}
```

Is there a recommended way of doing this in remark configuration? I realized that one way of doing this could be to add another remark configuration in the path which I want to alter, then require the base configuration, and add overrides, but I was wondering if there was a nicer way. 😃

**Comment by wooorm** [comment, marked as answer]
Indeed, adding a different config file closer to the differing files would do the trick. And if the config files are in JSON, you could require the “higher” config and add exceptions based on that data.

Overrides *is* a nice feature, and something like it becomes more important as unified is growing towards more file formats. But something like it doesn’t exist yet!

**Reply by silvenon** [reply]
> And if the config files are in JSON, you could require the “higher” config and add exceptions based on that data.

Could you expand on this? What I did is:

```js
exports.plugins = [
  ...require('path/to/.remarkrc').plugins,
  // overrides,
]
```

Are you saying that something like this is also possible in JSON?

**Reply by wooorm** [reply]
Sorry, that was quite confusing, I meant to write js*. Doet that clear it up?

**Reply by silvenon** [reply]
Yep! That sounds like what I'm doing 😃
